tmt/trying.py

```diff
@@ -32,85 +33,135 @@
+
+        for command in commands:
+            command_lower = command.lower()
+            if command_lower in Action._registry:
+                existing_action = Action._registry[command_lower]
+                raise ValueError(
+                    f"Command '{command}' is already registered for action "
+                    f"'{existing_action.full_name}' (function: {existing_action.func.__name__})"
+                )
+            Action._registry[command_lower] = self
 
-        return self.name.lower()
+    def __call__(self, func: ActionHandler) -> ActionHandler:
+        self.func = func
+        self.help_text = (
+            textwrap.dedent(func.__doc__).strip().splitlines()[0] if func.__doc__ else ""
+        ).lower()
+        return func
 
-    @property
-    def menu(self) -> str:
-        """
-        Show menu with the keyboard shortcut highlighted
-        """
+    @classmethod
+    def find(cls, command: str) -> 'Action':
+        action = cls._registry[command.lower()]
 
-        index = self.action.index(self.key)
+        # Do not expose hidden actions
+        if action.hidden:
+            raise KeyError("Hidden action")
 
-        before = style(self.action[0:index], fg="bright_blue")
-        key = style(self.key, fg="blue", bold=True, underline=True)
-        after = style(self.action[index + 1 :], fg="bright_blue")
+        return action
 
-        longest = max(len(action.name) for action in Action)
-        padding = " " * (longest + 3 - len(self.action))
+    @classmethod
+    def get_sorted_actions(cls) -> list['Action']:
+        """Get unique actions sorted by group and order"""
+        seen_functions: set[ActionHandler] = set()
+        actions: list[Action] = []
+        for action in sorted(
+            cls._registry.values(), key=lambda x: (x.group, x.order, x.func.__name__)
+        ):
+            if action.func not in seen_functions:
+                actions.append(action)
+                seen_functions.add(action.func)
+        return actions
+
+    @functools.cached_property
+    def primary_command(self) -> str:
+        """Return the primary command (the shortest command defined)"""
+        return min(self.commands, key=len)
+
+    @functools.cached_property
+    def key(self) -> str:
+        """Return the keyboard shortcut (first character of first command)"""
+        return self.primary_command[0]
 
-        return before + key + after + padding + self.description
+    @functools.cached_property
+    def full_name(self) -> str:
+        """Return the full command name (longest command)"""
+        return max(self.commands, key=len)
 
-    @classmethod
-    def find(cls, answer: str) -> "Action":
+    @functools.cached_property
+    def menu_item(self) -> str:
         """
-        Return action for given keyboard input (shortcut or whole word)
+        Show menu with the keyboard shortcut highlighted.
+        If the primary command is not a single letter, do not highlight anything.
```

Comment:
So... how is the user supposed to find out what is the shortcut when multiple letters are used?

Reply:
Maybe there is no shortcut? :)

Reply:
@psss yes, there is no shortcut when: 

1. there is no single letter shortcut among actions
2. it does not match anything in the long command

I am happy to adjust it as needed, it is a bit brittle, maybe we could say there always will be only actions with maximum 2 actions, one should be a single letter found in the string (error out otherwise?)